tox.ini

```diff
@@ -14,13 +14,20 @@ deps =
     pytest-xdist==1.15.0
     pytz==2016.7
     requests==2.11.1
-
-[testenv:desktop]
-commands = pytest --driver=Firefox {posargs:tests/desktop}
-
-[testenv:mobile]
-commands = pytest {posargs:tests/mobile}
+commands = pytest \
+    --junit-xml=results/{envname}.xml \
+    --html=results/{envname}.html \
+    {posargs:tests/{envname}}
```

Comment:
I had to remove `--driver=Firefox` because it was preventing me from overriding it via `PYTEST_ADDOPTS`. I've used `{envname}` to simplify to a single `[testenv]` section, as Tox will replace this with the active environment name. It does mean there's a environment->directory mapping, but in this case that's okay.